File: src/api/src/ImmoListing.Core/Models/Listing.cs
namespace ImmoListing.Core.Models;

public class Listing
{
    public long Id { get; set; }

    public string Name { get; set; }

    public PostalAddress PostalAddress { get; set; }

    public string Description { get; set; }

    public RealEstateListingBuildingType BuildingType { get; set; }

    public long LatestPriceEur { get; set; }

    public double SurfaceAreaM2 { get; set; }

    public int RoomsCount { get; set; }

    public int BedroomsCount { get; set; }

    public string ContactPhoneNumber { get; set; }

    public DateTime CreatedDate  { get; set; }
    
    public DateTime UpdatedDate  { get; set; }
}

File: src/client/src/types/Listing.ts
import { PostalAddress } from './PostalAddress';

export interface Listing {
  id: number;
  created_date: string;
  updated_date: string;
  name: string;
  postal_address: PostalAddress;
  description: string;
  building_type: string;
  latest_price_eur: number;
  surface_area_m2: string;
  rooms_count: string;
  bedrooms_count: string;
  contact_phone_number: string;
}
